src/OpenTelemetry/ReadOnlyTagCollection.cs

```diff
@@ -57,19 +57,6 @@ internal Enumerator(ReadOnlyTagCollection source)
         /// successfully advanced to the next element; <see
         /// langword="false"/> if the enumerator has passed the end of the
         /// collection.</returns>
-        public bool MoveNext()
-        {
-            int index = this.index;
-
-            if (index < this.source.Count)
-            {
-                this.Current = this.source.KeyAndValues[index];
-
-                this.index++;
-                return true;
-            }
-
-            return false;
-        }
+        public bool MoveNext() => ++this.index < this.source.Count;
```

Comment:
good change, but I'd always prefer if its handled on its own, as this is not related to Exemplars spec fixes.